### ISSUE TITLE
On a Big endian platform like AIX, KernelDef::CalculateHash() gives incorrect hash key

### ISSUE THREAD

**Issue by tvkai** [microsoft/onnxruntime#12728]
**Describe the bug**
A clear and concise description of what the bug is. To avoid repetition please make sure this is not one of the known issues mentioned on the respective release page.

KernelDef::CalculateHash() is generating a different hash key on big endian platform like AIX and this is leading to many tests
were failing.

**Urgency**
If there are particular important use cases blocked by this or strict project-related timelines, please share more information and dates. If there are no hard deadlines, please specify none.

**System information**
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): AIX
- ONNX Runtime installed from (source or binary): source
- ONNX Runtime version: latest
- Python version:
- Visual Studio version (if applicable):
- GCC/Compiler version (if compiling from source):
- CUDA/cuDNN version:
- GPU model and memory:

**To Reproduce**
- Describe steps/code to reproduce the behavior.
- Attach the ONNX model to the issue (where applicable) to expedite investigation.

Compile and Run onnxruntime on AIX.

**Expected behavior**
KernelDef::CalculateHash()  should generate same hash key.

Basically the issue is KernelDef::CalculateHash() is internally invoking  hash_int() -->  MurmurHash3::x86_128() with first parameter to an int. On big endian platforms this generates a different hash.

I tried the below fix ( after hash_str (op_name_);) in KernelDef::CalculateHash() , and it works on AIX .

void KernelDef::CalculateHash() {
  uint32_t hash[4] = {0, 0, 0, 0};

  auto hash_int = [&hash](int i) { MurmurHash3::x86_128(&i, sizeof(i), hash[0], &hash); };
  auto hash_str = [&hash](const std::string& str) {
    MurmurHash3::x86_128(str.data(), gsl::narrow_cast<int32_t>(str.size()), hash[0], &hash);
  };

  // use name, start/end, domain, provider and the type constraints.
  // we wouldn't have two kernels that only differed by the inplace or alias info or memory types.
  // currently nothing sets exec_queue_id either (and would assumably be a runtime thing and not part of the base
  // kernel definition)

  hash_str(op_name_);

  if constexpr (onnxruntime::endian::native == onnxruntime::endian::little) {
    hash_int(op_since_version_start_);
  } else {
    const uint8_t* c = (const uint8_t*)&op_since_version_start_;
    hash_int((uint32_t)c[0] |
           (uint32_t)c[1] << 8 |
           (uint32_t)c[2] << 16 |
           (uint32_t)c[3] << 24);
  }

 ----
 ----

**Screenshots**

**Additional context**
Add any other context about the problem here. If the issue is about a particular model, please share the model details as well to facilitate debugging.

**Comment by tvkai** [comment]
Submitted the changes.
